src/compiler/transformers/declarations.ts

```diff
@@ -677,15 +677,20 @@ namespace ts {
             return hasEffectiveModifier(node, ModifierFlags.Private) ? undefined : visitNodes(params, visitDeclarationSubtree);
         }
 
-        function isEnclosingDeclaration(node: Node) {
-            return isSourceFile(node)
-                || isTypeAliasDeclaration(node)
-                || isModuleDeclaration(node)
-                || isClassDeclaration(node)
-                || isInterfaceDeclaration(node)
-                || isFunctionLike(node)
-                || isIndexSignatureDeclaration(node)
-                || isMappedTypeNode(node);
+        function isEnclosingDeclaration(_node: Node) {
+            // TODO(jakebailey): Clean this up. If we look at the way that the "types"

```

Comment:
Just as an FYI, this list is the nodes our old declaration emitter captured as enclosing declarations while recurring, not many other reasons for it.

Reply:
Just to clarify, do you agree with removing this entirely or the opposite?

This is just a leftover now, I'll probably not touch this (I needed it in a previous iteration but haven't yet removed it, this branch is big WIP right now)

Reply:
There's no documented logic for why it's only *these* nodes which are/were captured - these are simply the only ones that were way back when the declaration emitter was string-based.